test/wanda_web/controllers/execution_controller_test.exs

```diff
@@ -61,4 +63,104 @@ defmodule WandaWeb.ExecutionControllerTest do
       end
     end
   end
+
+  describe "start execution" do
+    test "should start an execution", %{conn: conn} do
+      execution_id = UUID.uuid4()
+      group_id = UUID.uuid4()
+
+      json =
+        conn
+        |> put_req_header("content-type", "application/json")
+        |> post("/api/checks/executions/start", %{
+          "execution_id" => execution_id,
+          "group_id" => group_id,
+          "targets" => [
+            %{
+              "agent_id" => UUID.uuid4(),
+              "checks" => ["expect_check"]
+            }
+          ],
+          "env" => %{
+            "provider" => "azure"
+          }
+        })
+        |> json_response(202)
+
+      api_spec = ApiSpec.spec()
+      assert_schema(json, "AcceptedExecution", api_spec)
+    end
+
+    test "should return an error when attempting to start an already running execution", %{
+      conn: conn
+    } do
+      execution_id = UUID.uuid4()
+      group_id = UUID.uuid4()
+
+      targets = [
+        %{
+          agent_id: UUID.uuid4(),
+          checks: ["expect_check"]
+        }
+      ]
+
+      env = %{
+        "provider" => "azure"
+      }
+
+      # ?? start_supervised!
+      :ok =
+        Server.start_execution(
+          execution_id,
+          group_id,
+          Target.from_list(targets),
+          env
+        )
+
+      json =
+        conn
+        |> put_req_header("content-type", "application/json")
+        |> post("/api/checks/executions/start", %{
+          "execution_id" => execution_id,
+          "group_id" => group_id,
+          "targets" => targets,
+          "env" => env
+        })
+        |> json_response(422)
+
+      assert %{"error" => "already_running"} = json
+    end
+
+    test "should return an error on validation failure", %{conn: conn} do
+      json =
+        conn
+        |> put_req_header("content-type", "application/json")
+        |> post(
+          "/api/checks/executions/start",
+          %{
+            "group_id" => UUID.uuid4(),
```

Comment:
Maybe we could have a factory to create this test data. We already have the `env_factory` and `target_factory`

Reply:
Ok for the `env`, changed.

For the targets I cannot really use `target_factory` as it returns `Wanda.Executions.Target`.
What I need is transofrming plain maps to `Wanda.Executions.Target`

```elixir
targets = [
  %{
    agent_id: agent_id = UUID.uuid4(),
    checks: checks = ["expect_check"]
  }
]

# becomes

[
  %Target{
    agent_id: ^agent_id,
    checks: ^checks
  }
]
```

I'd keep it as it is and not create a factory that returns basically plain maps of targets
```elixir
[
  %{
    agent_id: agent_id = UUID.uuid4(),
    checks: checks = ["expect_check"]
  }
]
```

Works for you?